### PR TITLE
Package.swift: bump ArgumentParser to 1.1.4

### PR DESCRIPTION
**Pull request by MaxDesiatov**

### CODE REVIEW

**Comment by artemcm** [comment]
@MaxDesiatov this repository does not support cross-repo CI. 
Instead, it should be launched from `apple/swift` with the mention of this PR.

**Comment by MaxDesiatov** [comment]
Fair enough, thanks for the guidance 👍

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows

File: Utilities/build-script-helper.py [hunk 1]
```diff
@@ -333,9 +333,8 @@ def install_libraries(args, build_dir, universal_lib_dir, toolchain_lib_dir, tar
 
   # Install the swift-argument-parser shared libraries into the toolchain lib
   package_subpath = os.path.join(args.configuration, 'dependencies', 'swift-argument-parser')
-  for lib in ['libArgumentParser', 'libArgumentParserToolInfo']:
```
Comment (MaxDesiatov) [inline Utilities/build-script-helper.py:336]: @DougGregor this is a change I have doubts about, but it apparently makes the error in this CI run go away: https://ci.swift.org/job/swift-PR-macos/5203/console

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]
Looks good. Thanks for the extension fixes to `build-script-helper.py`

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by rauhul** [comment]
Just wondering, any reason to not pull in 1.2.0?

**Comment by MaxDesiatov** [comment]
There's a regression in 1.2.0 due to which SwiftPM bootstrap build fails.

**Comment by MaxDesiatov** [comment]
@swift-ci test Windows

**Comment by compnerd** [comment]
Bleh, the windows failure looks like a regression in LSP (wonder if the windows build got triggered on changes 🤔)

**Review by compnerd — CHANGES_REQUESTED** [review]
Windows does a shared link to s-a-p.  We should bump this in swift in update-checkout-config.json and in the other packages (SPM, LSP).

**Comment by MaxDesiatov** [comment]
It has been bumped in all other places, I have corresponding PRs waiting to be merged. Do your expect any additional changes to this PR specifically?

**Comment by compnerd** [comment]
Please test with following PRs:
apple/sourcekit-lsp#689

@swift-ci please test Windows platform

**Comment by MaxDesiatov** [comment]
I've been told upthread that this repository doesn't support cross-repo testing, but I hope that Windows CI run passes anyway

**Comment by MaxDesiatov** [comment]
@swift-ci test Windows

**Comment by MaxDesiatov** [comment]
@compnerd Windows build is passing now, would you mind re-reviewing? Thanks!

**Review by compnerd — APPROVED** [review]
The change itself looks fine, but please make sure that you do a cross-repo test in apple/swift with `@swift-ci please build toolchain Windows platform` and then test the resulting installer.  The problem is that there can be something subtle in the packaging where a dependency is either missing and expected or is missing and unexpected and the tools will not function.

**Comment by MaxDesiatov** [comment]
@compnerd thank you. I've pinged you in the corresponding PR, for some reason the Windows toolchain CI run can't find artifacts, even though there are no obvious failures. I've shared the error message here https://github.com/apple/swift/pull/59009#issuecomment-1371308975

**Comment by compnerd** [comment]
@MaxDesiatov - that is what happens when the necessary changes are missing!  You need a parallel change to apple/swift-installer-scripts: https://github.com/apple/swift/pull/59009#pullrequestreview-1235971354

**Comment by MaxDesiatov** [comment]
That did indeed help, thanks for the review and finding the root cause of the build failure!

**Pull request merged**